pkg/v21/resource/appmigration/create.go

```diff
@@ -66,17 +67,25 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, project.Name()),
 	}
 
+	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+	defer cancel()
```

Comment:
We use a short timeout and cancel the resource when it's exceeded. The migration is eventually consistent so we can just retry.

After 1 hour if the migration is not complete we get alerts because the chartconfig CRs are cordoned. So we can investigate.